doc/core/grpc-endpoint.md

```diff
@@ -0,0 +1,67 @@
+# gRPC Endpoints (`grpc_endpoint`) - TCP endpoint
+_Author: Sree Kuchibhotla (@sreecha) - Sep 2018_
+
+This document talks about `grpc_endpoint` abstraction and talks about the TCP endpoint's posix implementation in detail.
+
+### What is an endpoint ?
+`grpc_endpoint` is an abstraction used by gRPC to read/write bytes. There are different endpoint implementations depending on the underlying transport.
+
+
+### Endpoint interface
+
+The interface is documented well here: [`endpoint.h`](https://github.com/grpc/grpc/blob/v1.15.1/src/core/lib/iomgr/endpoint.h)
+
+### TCP endpoint
+
+TCP endpoint is implemented here: [tcp_posix.cc](https://github.com/grpc/grpc/blob/v1.15.1/src/core/lib/iomgr/tcp_posix.cc)
+
+#### Read
+
+``` C++
+// Implementation of grpc_endpoint_read() API
+void tcp_read(grpc_endpoint* ep, grpc_slice_buffer* slices, grpc_closure* cb) {
+ // 1. fd = Get grpc_fd from endpoint
+ // 2. Store read completion callback “cb” on the endpoint
+ // 3. Call polling engine API grpc_fd_notify_on_read(fd, tcp_handle_read)
```

Comment:
hmm this is not completely correct. We might directly call tcp_handle_read if it is not the first read

Reply:
This is supposed to be some high level detail on how the overall flow works...sort of a first order approximation.  Presenting more details here (but not ALL details) will only raise more questions in the readers mind...

Reply:
Presenting incomplete details is also pretty confusing imo, but we can always improve the documentation later I guess

Reply:
ok.. I will close this PR. On a second thought, It is not adding much value anyway.

Reply:
documentation is always nice